src/main/haskell/kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -113,3 +157,44 @@ evalKEq true false tools substitutionSimplifier _ pat =
       where
         ep1 = ExpandedPattern.fromPurePattern t1
         ep2 = ExpandedPattern.fromPurePattern t2
+
+evalKIte
+    ::  forall variable
+    .   ( FreshVariable variable
+        , Hashable variable
+        , OrdMetaOrObject variable
+        , SortedVariable variable
+        , ShowMetaOrObject variable
+        )
+    => MetadataTools.MetadataTools Object StepperAttributes
+    -> PredicateSubstitutionSimplifier Object Simplifier
+    -> PureMLPatternSimplifier Object variable
+    -> Application Object (PureMLPattern Object variable)
+    -> Simplifier
+        ( AttemptedFunction Object variable
+        , SimplificationProof Object
+        )
+evalKIte _ _ _ pat =
+    case pat of
+        Application
+            {  applicationChildren = [expr, t1, t2] } ->
+            evalIte expr t1 t2
+        _ -> Builtin.wrongArity "KEQUAL.ite"
+  where
+    evaluate :: Functor.Foldable.Fix (Pattern Object variable) -> Maybe Bool
+    evaluate expr = case Functor.Foldable.project <$> [expr] of
+        [DomainValuePattern a] -> Just (get a)
+        _                      -> Nothing
+
+    get :: DomainValue Object (BuiltinDomain child) -> Bool
+    get =
+        Builtin.runParser "KEQUAL.ite"
+        . Builtin.parseDomainValue Bool.parse
+
+    evalIte expr t1 t2 =
+        case evaluate expr of
+            Just result ->
+                if result
+                    then purePatternFunctionEvaluator t1
+                    else purePatternFunctionEvaluator t2
```

Comment:
I'm nitpicking, but I think this is more readable:
```
            Just True -> purePatternFunctionEvaluator t1
            Just False -> purePatternFunctionEvaluator t2
```

Reply:
It's generally not a good idea to have backtracking `case` statements like that; this would avoid backtracking:
```.hs
Just result
    | result -> purePatternFunctionEvaluator t1
    | otherwise -> purePatternFunctionEvaluator t2
```

Reply:
Oh, ok. Is this an efficiency concern? Or a readability concern?